### PR TITLE
fixed db_remove() if db is still in use.

### PR DESCRIPTION
**Pull request by guettli**
### What does this PR do?

fix db_remove() if db is still in use.

### What issues does this PR fix or reference?

postgres_database.absent reports the wrong error if the database is in use:  https://github.com/saltstack/salt/issues/42833

### CODE REVIEW

**Comment by guettli** [comment]
Could someone please review this pull request?

**Comment by DmitryKuzmenko** [comment]
@guettli can we add a test for this?
Also could you please fix lint errors and the existing tests errors?

File: salt/modules/postgres.py [hunk 1]
```diff
@@ -634,18 +633,19 @@ def db_remove(name, user=None, host=None, port=None, maintenance_db=None,
...
+        ret = _psql_prepare_and_run(['-c', query],
                                 user=user,
                                 host=host,
                                 port=port,
                                 runas=runas,
                                 maintenance_db=maintenance_db,
                                 password=password)
-    return ret['retcode'] == 0
-
+        if  ret['retcode'] != 0:
```
Comment (DmitryKuzmenko) [inline salt/modules/postgres.py]: Double spaces before `ret`.

File: salt/modules/postgres.py [hunk 2]
```diff
@@ -622,7 +622,6 @@ def db_alter(name, user=None, host=None, port=None, maintenance_db=None,
 
     return True
 
-
```
Comment (DmitryKuzmenko) [inline salt/modules/postgres.py]: PEP8 requires 2 blank lines here.

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]

**Comment by guettli** [comment]
I have no clue how to test this. If you could please provide some guidance. A link to an existing test might be enough.

**Comment by DmitryKuzmenko** [comment]
@guettli sorry for delay.
I think it's enough to:
1. Fix the existing test unit test `unit.modules.test_postgres.PostgresTestCase.test_db_remove`.
2. Add one more test very similar to `unit.modules.test_postgres.PostgresTestCase.test_db_remove` but checking the "error" case.

**Comment by guettli** [comment]
Do you mean first number one, then number two. Or do you mean do number one OR number two?

**Comment by guettli** [comment]
Sorry, I am busy. I think this is an internship tasks. Saltstack is company. Don't you have a some students for this?

**Review by DmitryKuzmenko — APPROVED** [review]

**Comment by DmitryKuzmenko** [comment]
re-run full amazon2-py2
re-run full macosxmojave-py2
re-run full centos7-py3-proxy

**Comment by DmitryKuzmenko** [comment]
re-run full macosxmojave-py2

**Comment by DmitryKuzmenko** [comment]
re-run full macosxmojave-py2

**Comment by DmitryKuzmenko** [comment]
re-run full macosxmojave-py2

**Comment by DmitryKuzmenko** [comment]
Py2 ci builds are isabled so this PR is green now. There's nothing to update to force rebuild all here. So I'm marking this as merge ready.

**Comment by DmitryKuzmenko** [comment]
@guettli could you please resolve the merge conflicts and run the pre-commit script on your code to re-format it with `isort` and `black` we're using now.
```
pip install pre-commit
pre-commit install
FILES="$(git show --pretty="" --name-only master..HEAD)"
pre-commit run isort $FILES
pre-commit run black $FILES
```
next time you'll commit it will be ran automatically.

**Comment by guettli** [comment]
@DmitryKuzmenko sorry, I can't reformat it. I switched the job and don't use salt any more. Sorry.

**Comment by DmitryKuzmenko** [comment]
@guettli thank you for clarification. I'll take care of this. Thank you for contribution

**Comment by DmitryKuzmenko** [comment]
Closing in favor of #56631.

**Pull request closed without merging**